tests/unit/amo/components/TestUserProfileEdit.js

```diff
@@ -0,0 +1,329 @@
+
+    expect(root.find('.UserProfileEdit-email')).toHaveLength(1);
+    expect(root.find('.UserProfileEdit-email')).toHaveProp('disabled', true);
+  });
+
+  it('renders a help text for the email and homepage fields', () => {
+    const root = renderUserProfileEdit({ params: { username: 'tofumatt' } });
+
+    expect(root.find('.UserProfileEdit--help')).toHaveLength(2);
+  });
+
+  it('renders a displayName input field', () => {
+    const root = renderUserProfileEdit({ params: { username: 'tofumatt' } });
+
+    expect(root.find('.UserProfileEdit-displayName')).toHaveLength(1);
+  });
+
+  it('renders a disabled displayName input field when not ready', () => {
+    const root = renderUserProfileEdit({ params: { username: 'not-ready' } });
+
+    expect(root.find('.UserProfileEdit-displayName'))
+      .toHaveProp('disabled', true);
+  });
+
+  it('renders a homepage input field', () => {
+    const root = renderUserProfileEdit({ params: { username: 'tofumatt' } });
+
+    expect(root.find('.UserProfileEdit-homepage')).toHaveLength(1);
+  });
+
+  it('renders a disabled homepage input field when not ready', () => {
+    const root = renderUserProfileEdit({ params: { username: 'not-ready' } });
+
+    expect(root.find('.UserProfileEdit-homepage')).toHaveProp('disabled', true);
+  });
+
+  it('renders a location input field', () => {
+    const root = renderUserProfileEdit({ params: { username: 'tofumatt' } });
+
+    expect(root.find('.UserProfileEdit-location')).toHaveLength(1);
+  });
+
+  it('renders a disabled location input field when not ready', () => {
+    const root = renderUserProfileEdit({ params: { username: 'not-ready' } });
+
+    expect(root.find('.UserProfileEdit-location')).toHaveProp('disabled', true);
+  });
+
+  it('renders a occupation input field', () => {
+    const root = renderUserProfileEdit({ params: { username: 'tofumatt' } });
+
+    expect(root.find('.UserProfileEdit-occupation')).toHaveLength(1);
+  });
+
+  it('renders a disabled occupation input field when not ready', () => {
+    const root = renderUserProfileEdit({ params: { username: 'not-ready' } });
+
+    expect(root.find('.UserProfileEdit-occupation')).toHaveProp('disabled', true);
+  });
+
+  it('renders a biography input field', () => {
+    const root = renderUserProfileEdit({ params: { username: 'tofumatt' } });
+
+    expect(root.find('.UserProfileEdit-biography')).toHaveLength(1);
+  });
+
+  it('renders a disabled biography input field when not ready', () => {
+    const root = renderUserProfileEdit({ params: { username: 'not-ready' } });
+
+    expect(root.find('.UserProfileEdit-biography')).toHaveProp('disabled', true);
+  });
+
+  it('dispatches editUserAccount action with all fields', () => {
+    const { store } = syncPropsAndParamsUsername('tofumatt');
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const errorHandler = createStubErrorHandler();
+
+    const root = mountUserProfileEdit({ errorHandler, store });
```

Comment:
Why do we need to use `mount` for this test? Is it because of the approach that was taken that binds the form fields to the component's props? I don't think we had to use `mount` in the tests for the edit collection form.

Reply:
`mount` is gone now :)